src/main/java/javax/validation/Configuration.java

```diff
@@ -127,6 +129,19 @@
 	T clockProvider(ClockProvider clockProvider);
 
 	/**
+	 * Adds a value extractor. Has priority over any extractors for the same
+	 * type and type parameter detected through the service loader or given in
+	 * the XML configuration.
+	 *
+	 * @param extractor value extractor implementation
+	 * @return {@code this} following the chaining method pattern.
+	 * @throws IllegalArgumentException If more than one extractor for the same
```

Comment:
I have mixed feelings about this. I think I would lean towards not throwing an exception and apply the last one defined. This way you are able to adapt a common configuration which might be defined in a util. Sure you could use XML but I wouldn't like using it just for that. Not a strong opinion though, just a feeling I thought better to express.

As for XML, I would agree that it doesn't make sense to allow this.

Agreed on the service loader part.